packages/background/src/utils/swaps/1inch.ts

```diff
@@ -106,13 +119,129 @@ export interface OneInchSwapRequestResponse {
     toToken: BasicToken;
     fromTokenAmount: string; // Input amount of fromToken in minimal divisible units
     toTokenAmount: string; // Result amount of toToken in minimal divisible units
-    protocols: unknown[]; // Route of the trade
-    tx: {
-        from: string; // Transactions will be sent from this address
-        to: string; // Transactions will be sent to our contract address
-        data: string; // Bytes of call data
-        value: string; // Amount of ETH (in wei) will be sent to the contract address
-        gas: number; // Estimated amount of the gas limit, increase this value by 25%
-        gasPrice: string; // Gas price in wei
-    };
+    protocols?: unknown[]; // Route of the trade
+    tx: SwapTxMeta;
+}
+
+export interface SwapTxMeta {
+    from: string; // Transactions will be sent from this address
+    to: string; // Transactions will be sent to our contract address
+    data: string; // Bytes of call data
+    value: string; // Amount of ETH (in wei) will be sent to the contract address
+    gas: number; // Estimated amount of the gas limit, increase this value by 25%
+    gasPrice: string; // Gas price in wei
 }
+
+export const OneInchService = {
+    parseQuoteParams({
+        fromToken,
+        toToken,
+        amount,
+    }: SwapQuoteParams): OneInchSwapQuoteParams {
+        return {
+            fromTokenAddress:
+                fromToken.address === '0x0'
+                    ? SWAP_NATIVE_ADDRESS
+                    : fromToken.address,
+            toTokenAddress:
+                toToken.address === '0x0'
+                    ? SWAP_NATIVE_ADDRESS
+                    : toToken.address,
+            amount,
+        };
+    },
+    async getSpender(chainId: number): Promise<string> {
+        try {
+            // 1Inch router contract address
+            const res = await retryHandling<OneInchSpenderRes>(() =>
+                httpClient.request<OneInchSpenderRes>(
+                    `${ONEINCH_SWAPS_ENDPOINT}${chainId}/approve/spender`
+                )
+            );
+
+            return res.address;
+        } catch (error) {
+            throw new Error('Unable to fetch exchange spender');
+        }
+    },
+    async getSwapQuote(
+        chainId: number,
+        params: SwapQuoteParams
+    ): Promise<SwapQuoteResponse> {
+        try {
+            const { fromTokenAddress, toTokenAddress, amount } =
+                this.parseQuoteParams(params);
+            const res = await retryHandling<OneInchSwapQuoteResponse>(() =>
+                httpClient.request<OneInchSwapQuoteResponse>(
+                    `${ONEINCH_SWAPS_ENDPOINT}${chainId}/quote`,
+                    {
+                        params: {
```

Comment:
REFERRER_ADDRESS is not needed in this quote, right?

Reply:
it's a read-only quote method so there's no need for the referrer param. Both APIs require it when you are requesting the actual swap transaction to build before presenting it to the user to sign, that way they can present the TX data with our referrer address and fee.